assets/lokomotive-kubernetes/bootkube/resources/charts/kube-apiserver/templates/kube-apiserver.yaml

```diff
@@ -1,20 +1,53 @@
+# On platforms like AWS, it's enough for kube-apiserver to listen on Node InternalIP. On other
+# like Packet, we need to listen on all interfaces to be able to expose kube-apiserver to the
+# internet. As we cannot listen on specific IP address and on 0.0.0.0 on the same port when using
+# SO_REUSEPORT, kube-apiserver process must listen on different port than HAProxy does. So if we
+# need to expose kube-apiserver on all interfaces, kube-apiserver will be listening on random
+# 127.0.0.0/8 IP address on port 7443, but will advertise Node InternalIP with port 7443 and HAProxy
+# will be listening on both Node InternalIP on port 7443 and on 0.0.0.0:6443 for both internal and
+# external traffic. See https://github.com/kinvolk/lokomotive/pull/32 for more details.
+{{ define "port" }}
+{{- if and .Values.apiserver.exposeOnAllInterfaces (eq (int .Values.apiserver.replicas) 1) -}}
+7443
+{{- else -}}
+6443
+{{- end -}}
+{{ end }}
 apiVersion: apps/v1
+# If there is jsut one controller node, we want to use Deployment to be able to run 2 kube-apiserver
+# pods on a single node at a time, to provide graceful upgrades.
+# See https://github.com/kinvolk/lokomotive/pull/32 for more details.
```

Comment:
This link doesn't seem very useful, maybe point to a specific comment or something?

Reply:
I mentioned to look for commit messages and comments. Does this make more sense?

Reply:
I think mentioning the commit that explains this issue in a lot detail is way better (as I commented here: https://github.com/kinvolk/lokomotive/pull/32#discussion_r390502037). Does it make sense for you too?

Reply:
Ok, will do.

Reply:
Done.